app/lib/package/backend.dart

```diff
@@ -824,7 +825,23 @@ Future<void> _verifyTarball(String filename) async {
   if (fileSize == 0) {
     throw PackageRejectedException.archiveEmpty();
   }
-  // TODO: check if the file is valid .tar.gz
+  // Check if the file has any symlink.
+  final pr = await runProc('tar', ['-tvf', filename]);
+  if (pr.exitCode != 0) {
+    _logger.info('Rejecting package: tar returned with ${pr.exitCode}\n'
+        '${pr.stdout}\n${pr.stderr}');
+    throw PackageRejectedException.invalidTarGz();
+  }
+  final lines = pr.stdout.toString().split('\n');
+  for (final line in lines) {
+    if (line.startsWith('l')) {
+      // report only the source path
+      // if output is non-standard for any reason, this reports the full line
+      final parts = line.split(' -> ');
+      final source = parts.first.split(' ').last;
+      throw PackageRejectedException.containsSymlink(source);
+    }
+  }
 }
```

Comment:
Can we put this in a small function somewhere and test it... just for sanity?

Reply:
The happy path is tested by integration tests, so I assume you are looking for a test where we create an archive with symlinks? or use one that is known to have it?

Reply:
Yes - testing that this fails with a symlink would be nice!